sdk/src/main/java/com/mapbox/maps/renderer/widget/BitmapWidgetRenderer.kt

```diff
@@ -0,0 +1,311 @@
+    vertexPositionBuffer.put(
+      -bitmapWidth / 2f, -bitmapHeight / 2f,
+      -bitmapWidth / 2f, bitmapHeight / 2f,
+      bitmapWidth / 2f, -bitmapHeight / 2f,
+      bitmapWidth / 2f, bitmapHeight / 2f,
+    )
+  }
+
+  private fun topY() = when (position.vertical) {
+    WidgetPosition.Vertical.BOTTOM -> surfaceHeight.toFloat() - bitmapHeight.toFloat() / 2f - marginY
+    WidgetPosition.Vertical.CENTER -> surfaceHeight.toFloat() / 2 - bitmapHeight.toFloat() / 2f + marginY
+    WidgetPosition.Vertical.TOP -> marginY + bitmapHeight.toFloat() / 2f
+  }
+
+  private fun leftX() = when (position.horizontal) {
+    WidgetPosition.Horizontal.LEFT -> marginX + bitmapWidth.toFloat() / 2f
+    WidgetPosition.Horizontal.CENTER -> surfaceWidth.toFloat() / 2 - bitmapWidth.toFloat() / 2f + marginX
+    WidgetPosition.Horizontal.RIGHT -> surfaceWidth.toFloat() - bitmapWidth.toFloat() / 2f - marginX
+  }
+
+  override fun prepare() {
+    val maxAttrib = IntArray(1)
+    GLES20.glGetIntegerv(GLES20.GL_MAX_VERTEX_ATTRIBS, maxAttrib, 0)
+
+    vertexShader = GlUtils.loadShader(
+      GLES20.GL_VERTEX_SHADER,
+      VERTEX_SHADER_CODE
+    ).also(GlUtils::checkCompileStatus)
+
+    fragmentShader = GlUtils.loadShader(
+      GLES20.GL_FRAGMENT_SHADER,
+      FRAGMENT_SHADER_CODE
+    ).also(GlUtils::checkCompileStatus)
+
+    program = GLES20.glCreateProgram().also { program ->
+      GlUtils.checkError("glCreateProgram")
+
+      GLES20.glAttachShader(program, vertexShader)
+      GlUtils.checkError("glAttachShader")
+
+      GLES20.glAttachShader(program, fragmentShader)
+      GlUtils.checkError("glAttachShader")
+
+      GLES20.glLinkProgram(program)
+      GlUtils.checkError("glLinkProgram")
+    }
+
+    uniformMvpMatrix =
+      GLES20.glGetUniformLocation(program, "uMvpMatrix")
+    GlUtils.checkError("glGetUniformLocation")
+
+    attributeVertexPosition =
+      GLES20.glGetAttribLocation(program, "aPosition")
+    GlUtils.checkError("glGetAttribLocation")
+
+    attributeTexturePosition =
+      GLES20.glGetAttribLocation(program, "aCoordinate")
+    GlUtils.checkError("glGetAttribLocation")
+
+    uniformTexture =
+      GLES20.glGetUniformLocation(program, "uTexture")
+    GlUtils.checkError("glGetUniformLocation")
+
+    needRender = true
+  }
+
+  override fun render() {
+    if (program == 0) {
+      prepare()
+    }
+    GLES20.glUseProgram(program)
+    GlUtils.checkError("glUseProgram")
+
+    if (updateMatrix) {
+      Matrix.setIdentityM(mvpMatrix, 0)
+
+      Matrix.multiplyMM(translateRotate, 0, translateMatrix, 0, rotationMatrix, 0)
+      Matrix.multiplyMM(mvpMatrix, 0, screenMatrix, 0, translateRotate, 0)
+
+      mvpMatrixBuffer.rewind()
+      mvpMatrixBuffer.put(mvpMatrix)
+      mvpMatrixBuffer.rewind()
+
+      updateMatrix = false
+    }
+
+    GLES20.glUniformMatrix4fv(uniformMvpMatrix, 1, false, mvpMatrixBuffer)
+
+    textureFromBitmap()
```

Comment:
it's static image drawer, correct? why can't we load texture once from the bitmap?